crates/wasmtime/src/runtime/vm/const_expr.rs

```diff
@@ -73,7 +73,19 @@ impl<'a> ConstEvalContext<'a> {
             .collect::<Vec<_>>();
 
         let allocator = StructRefPre::_new(store, struct_ty);
-        let struct_ref = StructRef::_new(store, &allocator, &fields)?;
+        let struct_ref = match StructRef::_new(store, &allocator, &fields) {
+            Ok(s) => s,
+            Err(e) => {
+                if let Some(oom) = e.downcast_ref::<crate::GcHeapOutOfMemory<()>>() {
+                    store
+                        .traitobj_mut()
+                        .maybe_async_grow_or_collect_gc_heap(oom.bytes_needed())?;
```

Comment:
To double-check, you've confirmed that all locations that evaluate const exprs are on fibers?

Reply:
Yep :+1: